File: src/main/java/ua/borovyk/catalogue/data/entity/ProductImage.java
package ua.borovyk.catalogue.data.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
@ToString
@Entity
@Table(name = "product_image")
public class ProductImage {

    @Id
    @GeneratedValue
    private Long id;

    @Column(nullable = false)
    private String path;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String extension;

    @Column(name = "order_index", nullable = false)
    private Integer orderIndex;

    @ManyToOne(optional = false)
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

}

File: src/main/ui/src/models/ProductImage.js
import {Model} from "vue-mc";

export default class ProductImage extends Model {

    defaults () {
        return {
            id: null,
            path: null,
            name: null,
            extension: null,
            productId: null,
        }
    }

    mutations () {
        return {
            id: (id) => Number(id) || null,
            path: (path) => String(path) || null,
            name: (name) => String(name) || null,
            extension: (extension) => String(extension) || null,
            productId: (productId) => Number(productId) || null,
        }
    }
}
